core/src/test/java/com/epam/esm/service/impl/GiftServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.GiftDao;
+import com.epam.esm.dao.GiftTagDao;
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.impl.GiftDaoImpl;
+import com.epam.esm.dao.impl.GiftTagDaoImpl;
+import com.epam.esm.dao.impl.TagDaoImpl;
+import com.epam.esm.model.CustomSearchRequest;
+import com.epam.esm.model.dto.GiftCertificateDto;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+import com.epam.esm.service.GiftService;
+import com.epam.esm.util.SearchConstants;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class GiftServiceImplTest {
+    private static GiftDao giftDao;
+    private static TagDao tagDao;
+    private static GiftTagDao giftTagDao;
+    private static GiftService giftService;
+
+    private Timestamp currentTimestamp;
+    private List<GiftCertificateEntity> giftCertificateEntityList;
+    private GiftCertificateEntity giftCertificateEntity;
+
+    @BeforeAll
+    static void setup() {
+        giftDao = mock(GiftDaoImpl.class);
+        tagDao = mock(TagDaoImpl.class);
+        giftTagDao = mock(GiftTagDaoImpl.class);
+        giftService = new GiftServiceImpl(giftDao, tagDao, giftTagDao);
+    }
+
+    @BeforeEach
+    void init() {
+        currentTimestamp = new Timestamp(System.currentTimeMillis());
+        giftCertificateEntityList = new ArrayList<>();
+        for (long i = 1; i <= 5; i++) {
+            giftCertificateEntityList.add(GiftCertificateEntity.builder()
+                    .id(i)
+                    .name("name" + i)
+                    .description("description" + i)
+                    .price((int) i)
+                    .duration((int) i)
+                    .createDate(currentTimestamp)
+                    .lastUpdateDate(currentTimestamp).build());
+        }
+        giftCertificateEntity = GiftCertificateEntity.builder()
```

Comment:
why are you creating this object when you can just get the same one form list created above?